File: sistema_examenes_backend/src/main/java/com/sistema/examenes/Services/PreguntaService.java
package com.sistema.examenes.Services;

import com.sistema.examenes.Models.Examen;
import com.sistema.examenes.Models.Pregunta;

import java.util.Set;

public interface PreguntaService {

    Pregunta agregarPregunta(Pregunta pregunta);
    Pregunta actualizarPregunta(Pregunta pregunta);
    Set<Pregunta> obtenerPreguntas();
    Pregunta obtenerPregunta(Long pregunta_id);
    Set<Pregunta> obtenerPreguntasDelExamen(Examen examen);
    void eliminarPregunta(Long preguntaId);
}

File: sistema_examenes_frontend/src/app/services/pregunta.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class PreguntaService {

  constructor(private http:HttpClient ) { }

  public listarPreguntasDelExamen(examenId:any){
    return this.http.get(`${baseUrl}/preguntas/examen/todos/${examenId}`)
  }

  public guardarPregunta(pregunta:any){
    return this.http.post(`${baseUrl}/preguntas/`, pregunta)
  }

  public eliminarPregunta(preguntaId:any){
    return this.http.delete(`${baseUrl}/preguntas/${preguntaId}`)
  }

  public actualizarPregunta(pregunta:any){
    return this.http.put(`${baseUrl}/preguntas/`, pregunta)
  }

  public obtenerPregunta(preguntaId:any){
    return this.http.get(`${baseUrl}/preguntas/${preguntaId}`)
  }
}
